source/common/upstream/outlier_detection_impl.cc

```diff
@@ -214,18 +219,23 @@ void EventLoggerImpl::logEject(HostDescriptionPtr host, EjectionType type) {
   static const std::string json =
     std::string("{{") +
     "\"time\": \"{}\", " +
+    "\"secs_since_last_action\": \"{}\", " +
     "\"cluster\": \"{}\", " +
     "\"upstream_url\": \"{}\", " +
     "\"action\": \"eject\", " +
     "\"type\": \"{}\", " +
     "\"num_ejections\": {}" +
     "}}\n";
   // clang-format on
-
-  file_->write(fmt::format(json,
-                           AccessLogDateTimeFormatter::fromTime(time_source_.currentSystemTime()),
-                           host->cluster().name(), host->address()->asString(), typeToString(type),
-                           host->outlierDetector().numEjections()));
+  SystemTime now = time_source_.currentSystemTime();
+  file_->write(
+      fmt::format(json, AccessLogDateTimeFormatter::fromTime(now),
+                  ((host->outlierDetector().lastUnejectionTime().valid())
+                       ? (std::chrono::duration_cast<std::chrono::seconds>(
```

Comment:
can you move this to a little helper method and reuse it everywhere? (checking for valid, return -1, etc)